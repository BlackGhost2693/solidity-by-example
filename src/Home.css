.Home-header {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.Home-logo {
  width: 150px;
}
