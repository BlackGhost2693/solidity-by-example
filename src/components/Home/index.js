import React from "react"
import PropTypes from "prop-types"
import SEO from "../SEO"
import Label from "../Label"
import styles from "./index.module.css"

function Home(props) {
  const { version, routesByCategory } = props

  return (
    <div className={styles.component}>
      <SEO
        title={`Solidity by Example | ${version}`}
        description="Learn smart contract programming using Solidity"
      />
      <h1 className={styles.header}>
        <a href="/">Solidity by Example {version}</a>
      </h1>
      <div className={styles.main}>
        <p>
          an introduction to{" "}
          <a href="https://solidity.readthedocs.io">Solidity</a> with simple
          examples
        </p>

        {routesByCategory.map(({ routes, title }, i) => (
          <div key={i}>
            {title && <h3>{title}</h3>}

            <ul className={styles.list}>
              {routes.map(({ path, title, breakingChanges }) => (
                <li className={styles.listItem} key={path}>
                  <a href={path}>{title}</a>
                  {breakingChanges && (
                    <div className={styles.label}>
                      <Label />
                    </div>
                  )}
                </li>
              ))}
            </ul>
          </div>
        ))}
      </div>
    </div>
  )
}

Home.propTypes = {
  version: PropTypes.oneOf(["0.6.0", "0.5.16"]).isRequired,
  routesByCategory: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      routes: PropTypes.arrayOf(
        PropTypes.shape({
          path: PropTypes.string.isRequired,
          title: PropTypes.string.isRequired,
          breakingChanges: PropTypes.bool,
        })
      ).isRequired,
    })
  ).isRequired,
}

export default Home
