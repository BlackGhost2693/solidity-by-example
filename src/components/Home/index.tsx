import React from "react"
import SEO from "../SEO"
import Label from "../Label"
import styles from "./index.module.css"

interface Route {
  path: string
  title: string
  breakingChanges?: boolean
}

interface Props {
  version: "0.6.10" | "0.5.16"
  routesByCategory: { title: string; routes: Route[] }[]
}

const Home: React.FC<Props> = ({ version, routesByCategory }) => {
  return (
    <div className={styles.component}>
      <SEO
        title={`Solidity by Example | ${version}`}
        description="Learn smart contract programming using Solidity"
      />
      <h1 className={styles.header}>
        <a href="/">Solidity by Example</a>
      </h1>
      <div className={styles.subHeader}>v {version}</div>
      <div className={styles.main}>
        <p>
          an introduction to{" "}
          <a href="https://solidity.readthedocs.io">Solidity</a> with simple
          examples
        </p>

        {routesByCategory.map(({ routes, title }, i) => (
          <div key={i}>
            {title && <h3>{title}</h3>}

            <ul className={styles.list}>
              {routes.map(({ path, title, breakingChanges }) => (
                <li className={styles.listItem} key={path}>
                  <a href={path}>{title}</a>
                  {breakingChanges && (
                    <div className={styles.label}>
                      <Label />
                    </div>
                  )}
                </li>
              ))}
            </ul>
          </div>
        ))}
      </div>
    </div>
  )
}

export default Home
