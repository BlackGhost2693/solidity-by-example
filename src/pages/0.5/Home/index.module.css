.component {
  max-width: 420px;
  /* background-color: green; */
}
.component ul {
  list-style: none;
  padding: 0;
}
.header a {
  text-decoration: none;
}
