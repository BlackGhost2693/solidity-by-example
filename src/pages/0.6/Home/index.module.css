.component {
  /* max-width: 420px; */
  /* background-color: green; */
}
.component ul {
  list-style: none;
  padding: 0;
}
.listItem {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-top: 10px;
  margin-bottom: 10px;
  white-space: nowrap;
}
.label {
  margin-left: 10px;
}
.header a {
  text-decoration: none;
}
