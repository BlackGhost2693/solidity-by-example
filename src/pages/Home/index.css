.Home {
  max-width: 420px;
  /* background-color: green; */
}
.Home-header a {
  text-decoration: none;
}
.Home ul {
  list-style: none;
  padding: 0;
}
