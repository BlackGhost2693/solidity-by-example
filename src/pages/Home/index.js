import React from "react"
import SEO from "../../components/SEO"
import "./index.css"

function Home() {
  return (
    <div className="Home">
      <SEO
        title="Solidity by Example"
        description="Learn smart contract programming using Solidity"
      />
      <h1 className="Home-header">
        <a href="/">Solidity by Example</a>
      </h1>
      <div className="Home-main">
        <p>
          an introduction to{" "}
          <a href="https://solidity.readthedocs.io">Solidity</a> using examples
        </p>

        <ul className="Home-list">
          <li>
            <a href="/hello-world">Hello World</a>
          </li>
          <li>
            <a href="/state-variables">State Variables</a>
          </li>
          <li>
            <a href="/ether-units">Ether and Wei</a>
          </li>
          <li>
            <a href="/gas">Gas and Gas Price</a>
          </li>
          <li>
            <a href="/function">Function</a>
          </li>
          <li>
            <a href="/view-and-pure-functions">View and Pure Functions</a>
          </li>
          <li>
            <a href="/function-modifier">Function Modifier</a>
          </li>
          <li>
            <a href="/constructor">Constructor</a>
          </li>
          <li>
            <a href="/inheritance">Inheritance</a>
          </li>
          <li>
            <a href="/super">Calling Parent Contracts</a>
          </li>
          <li>
            <a href="/shadowing-inherited-state-variables">
              Shadowing Inherited State Variables
            </a>
          </li>
          <li>
            <a href="/visibility">Visibility</a>
          </li>
          <li>
            <a href="/events">Events</a>
          </li>
          <li>
            <a href="/error">Error</a>
          </li>
          <li>
            <a href="/loop">Loop</a>
          </li>
          <li>
            <a href="/array">Array</a>
          </li>
          <li>
            <a href="/mapping">Mapping</a>
          </li>
          <li>
            <a href="/enum">Enum</a>
          </li>
          <li>
            <a href="/structs">Structs</a>
          </li>
          <li>
            <a href="/new-contract">Creating Contract from Other Contract</a>
          </li>
          <li>
            <a href="/calling-contract">Calling Other Contract</a>
          </li>
          <li>
            <a href="/call">Call</a>
          </li>
          <li>
            <a href="/delegatecall">Delegatecall</a>
          </li>
          <li>
            <a href="/Library">Library</a>
          </li>
          <li>
            <a href="/payable">Payable</a>
          </li>
          <li>
            <a href="/sending-ether">Sending Ether</a>
          </li>
          <li>
            <a href="/fallback">Fallback</a>
          </li>
          <li>
            <a href="/fallback-gas-limit">Fallback Gas Limit</a>
          </li>
          <li>
            <a href="/re-entrancy">Re-Entrancy</a>
          </li>
          <li>
            <a href="/self-destruct">Self Destruct</a>
          </li>
          <li>
            <a href="/hashing">Hashing with Keccak256</a>
          </li>
          <li>
            <a href="/signature">Verifying Signature</a>
          </li>
          <li>
            <a href="/import">Import</a>
          </li>
          <li>
            <a href="/safe-math">Safe Math</a>
          </li>
        </ul>

        <h3>Applications</h3>
        <ul>
          <li>
            <a href="/multi-sig-wallet">Multi-Sig Wallet</a>
          </li>
          <li>
            <a href="/dividend">Dividend</a>
          </li>
          <li>
            <a href="/payment-channel">PaymentChannel</a>
          </li>
        </ul>
      </div>
    </div>
  )
}

export default Home
