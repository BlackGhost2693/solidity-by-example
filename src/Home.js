import React, { Component } from 'react'
import './Home.css'
import Footer from './Footer'

class Home extends Component {
  render() {
    return (
      <div className="Home">
        <h1 className="Home-header">
          <a href="/">Solidity by Example</a>
        </h1>
        <div className="Home-main">
          <p>
            an introduction to <a href="https://solidity.readthedocs.io">Solidity</a> using example programs
          </p>

          <ul className="Home-list">
            <li><a href="/hello-world">Hello World</a></li>
            <li><a href="/simple-storage">Simple Storage</a></li>
            <li><a href="/function">Function</a></li>
            <li><a href="/function-modifier">Function Modifier</a></li>
            <li><a href="/events">Events</a></li>
            <li><a href="/error">Error</a></li>
            <li><a href="/payable">Payable</a></li>
            <li><a href="/sending-ether">Sending Ether</a></li>
            <li><a href="/fallback">Fallback</a></li>
            <li><a href="/fallback-gas-limit">Fallback Gas Limit</a></li>
            <li><a href="/re-entrancy">Re-Entrancy</a></li>
            <li><a href="/self-destruct">Self Destruct</a></li>
            <li><a href="/import">Import</a></li>
            <li><a href="/safe-math">Safe Math</a></li>
          </ul>
        </div>

        <Footer />
      </div>
    )
  }
}

export default Home
