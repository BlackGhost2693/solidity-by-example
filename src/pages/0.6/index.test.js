import React from "react"
import { shallow } from "enzyme"
import Home from "./index"

test("it renders", () => {
  const component = shallow(<Home />)

  expect(component).toMatchSnapshot()
})
