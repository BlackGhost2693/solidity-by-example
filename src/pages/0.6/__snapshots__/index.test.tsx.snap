// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<Home
  routesByCategory={
    Array [
      Object {
        "routes": Array [
          Object {
            "path": "/0.6/hello-world",
            "title": "Hello World",
          },
          Object {
            "path": "/0.6/first-app",
            "title": "First App",
          },
          Object {
            "path": "/0.6/primitives",
            "title": "Primitive Data Types",
          },
          Object {
            "path": "/0.6/variables",
            "title": "Variables",
          },
          Object {
            "path": "/0.6/state-variables",
            "title": "Reading and Writing to a State Variable",
          },
          Object {
            "path": "/0.6/ether-units",
            "title": "Ether and Wei",
          },
          Object {
            "path": "/0.6/gas",
            "title": "Gas and Gas Price",
          },
          Object {
            "path": "/0.6/if-else",
            "title": "If / Else",
          },
          Object {
            "path": "/0.6/loop",
            "title": "For and While Loop",
          },
          Object {
            "path": "/0.6/mapping",
            "title": "Mapping",
          },
          Object {
            "path": "/0.6/array",
            "title": "Array",
          },
          Object {
            "path": "/0.6/enum",
            "title": "Enum",
          },
          Object {
            "path": "/0.6/structs",
            "title": "Structs",
          },
          Object {
            "path": "/0.6/data-locations",
            "title": "Data Locations - Storage, Memory and Calldata",
          },
          Object {
            "path": "/0.6/function",
            "title": "Function",
          },
          Object {
            "path": "/0.6/view-and-pure-functions",
            "title": "View and Pure Functions",
          },
          Object {
            "path": "/0.6/error",
            "title": "Error",
          },
          Object {
            "path": "/0.6/function-modifier",
            "title": "Function Modifier",
          },
          Object {
            "path": "/0.6/events",
            "title": "Events",
          },
          Object {
            "path": "/0.6/constructor",
            "title": "Constructor",
          },
          Object {
            "breakingChanges": true,
            "path": "/0.6/inheritance",
            "title": "Inheritance",
          },
          Object {
            "breakingChanges": true,
            "path": "/0.6/shadowing-inherited-state-variables",
            "title": "Shadowing Inherited State Variables",
          },
          Object {
            "path": "/0.6/super",
            "title": "Calling Parent Contracts",
          },
          Object {
            "path": "/0.6/visibility",
            "title": "Visibility",
          },
          Object {
            "path": "/0.6/interface",
            "title": "Interface",
          },
          Object {
            "path": "/0.6/payable",
            "title": "Payable",
          },
          Object {
            "path": "/0.6/sending-ether",
            "title": "Sending Ether - Transfer, Send, and Call",
          },
          Object {
            "breakingChanges": true,
            "path": "/0.6/fallback",
            "title": "Fallback",
          },
          Object {
            "path": "/0.6/call",
            "title": "Call",
          },
          Object {
            "path": "/0.6/delegatecall",
            "title": "Delegatecall",
          },
          Object {
            "path": "/0.6/calling-contract",
            "title": "Calling Other Contract",
          },
          Object {
            "path": "/0.6/new-contract",
            "title": "Creating Contracts from a Contract",
          },
          Object {
            "newFeature": true,
            "path": "/0.6/try-catch",
            "title": "Try / Catch",
          },
          Object {
            "path": "/0.6/import",
            "title": "Import",
          },
          Object {
            "path": "/0.6/library",
            "title": "Library",
          },
          Object {
            "path": "/0.6/hashing",
            "title": "Hashing with Keccak256",
          },
          Object {
            "path": "/0.6/signature",
            "title": "Verifying Signature",
          },
        ],
        "title": "",
      },
      Object {
        "routes": Array [
          Object {
            "path": "/0.6/app/multi-sig-wallet",
            "title": "Multi Sig Wallet",
          },
          Object {
            "path": "/0.6/app/merkle-tree",
            "title": "Merkle Tree",
          },
          Object {
            "path": "/0.6/app/iterable-mapping",
            "title": "Iterable Mapping",
          },
          Object {
            "path": "/0.6/app/erc20",
            "title": "ERC20",
          },
          Object {
            "path": "/0.6/app/create2",
            "title": "Precompute Contract Address with Create2",
          },
          Object {
            "path": "/0.6/app/uni-directional-payment-channel",
            "title": "Uni-directional Payment Channel",
          },
          Object {
            "path": "/0.6/app/bi-directional-payment-channel",
            "title": "Bi-directional Payment Channel",
          },
        ],
        "title": "Applications",
      },
      Object {
        "routes": Array [
          Object {
            "path": "/0.6/hacks/re-entrancy",
            "title": "Re-Entrancy",
          },
          Object {
            "path": "/0.6/hacks/overflow",
            "title": "Arithmetic Overflow and Underflow",
          },
          Object {
            "path": "/0.6/hacks/self-destruct",
            "title": "Self Destruct",
          },
          Object {
            "path": "/0.6/hacks/accessing-private-data",
            "title": "Accessing Private Data",
          },
          Object {
            "path": "/0.6/hacks/delegatecall",
            "title": "Delegatecall",
          },
          Object {
            "path": "/0.6/hacks/randomness",
            "title": "Source of Randomness",
          },
          Object {
            "path": "/0.6/hacks/denial-of-service",
            "title": "Denial of Service",
          },
          Object {
            "path": "/0.6/hacks/phishing-with-tx-origin",
            "title": "Phishing with tx.origin",
          },
          Object {
            "path": "/0.6/hacks/hiding-malicious-code-with-external-contract",
            "title": "Hiding Malicious Code with External Contract",
          },
          Object {
            "path": "/0.6/hacks/honeypot",
            "title": "Honeypot",
          },
          Object {
            "path": "/0.6/hacks/front-running",
            "title": "Front Running",
          },
          Object {
            "path": "/0.6/hacks/block-timestamp-manipulation",
            "title": "Block Timestamp Manipulation",
          },
          Object {
            "path": "/0.6/hacks/signature-replay",
            "title": "Signature Replay",
          },
        ],
        "title": "Hacks",
      },
    ]
  }
  version="0.6.10"
/>
`;
